guide/dev/code/submodules.md

```diff
@@ -5,31 +5,55 @@ layout: website-normal
 
 ## Setting Up Forks
 
-If you're contributing regularly, you'll want your submodules to pull from the `apache` repo
+If you're contributing regularly, you'll want your submodules to pull from the repo in the Apache GitHub org
 and push to your fork.
-You'll need to create a fork and the set up your local submodules should then have:
+You'll need to create a fork and then set up your local submodules should then have:
 
 * your fork as the `origin` remote repo
-* the Apache github repo as the `upstream` remote repo
-* optionally the repo at the Apache git server as the `apache` remote repo  
+* the Apache GitHub repo as the `upstream` remote repo, 
+  optionally with `origin` set as the `push` target (since you can't push to `github.com/apache`)
+  and fetch set up to get pull-request branches
+* optionally the repo at the Apache git server as the `apache-git` remote repo
 
-This can be done automatically with the following commands:
+Assuming you've checked out from the Apache GitHub repos 
+following [the standard instructions](index.html)
+(such that that repo is the `origin` remote and you don't have any forks in your GitHub account -- 
+if you do you should probably delete them as their upstream is likely to be wrong!),
+then you can use the following script to automatically do the fork at GitHub and modify your locally-defined remote repos:
 
 {% highlight bash %}
-    hub fork
-    git remote add apache https://git-wip-us.apache.org/repos/asf/$x
-    for x in brooklyn-* ; do
-      pushd $x
-      hub fork
-      git remote add apache https://git-wip-us.apache.org/repos/asf/$x
-      popd
-    done
+hub fork
+# the above triggers a fork at GitHub and then does a git remote rename of origin to upstream 
+# and a git add origin pointing to your fork
+
+# configure upstream so pushes go to origin
+git remote set-url --push upstream $(git remote -v | grep origin | grep push | awk '{print $2}')
```

Comment:
Hacking the `upstream` push URL to actually push to a different repository? I've never seen that before, and I don't think I like it - it's quite deceptive :confused:

Is this article on recently-added Git support for triangular workflows helpful? https://github.com/blog/2042-git-2-5-including-multiple-worktrees-and-triangular-workflows